src/main/scala/rocket/RocketCore.scala

```diff
@@ -876,12 +876,12 @@ class Rocket(tile: RocketTile)(implicit p: Parameters) extends CoreModule()(p)
   coreMonitorBundle.valid := csr.io.trace(0).valid && !csr.io.trace(0).exception
   coreMonitorBundle.pc := csr.io.trace(0).iaddr(vaddrBitsExtended-1, 0).sextTo(xLen)
   coreMonitorBundle.wren := wb_wen && !wb_set_sboard
-  coreMonitorBundle.wrdst := wb_waddr
+  coreMonitorBundle.wrdst := Mux(wb_ctrl.wxd || wb_ctrl.wfd, wb_waddr, 0.U)
   coreMonitorBundle.wrdata := Mux(coreMonitorBundle.wren, rf_wdata, 0.U)
-  coreMonitorBundle.rd0src := wb_reg_inst(19,15)
-  coreMonitorBundle.rd0val := Reg(next=Reg(next=ex_rs(0)))
-  coreMonitorBundle.rd1src := wb_reg_inst(24,20)
-  coreMonitorBundle.rd1val := Reg(next=Reg(next=ex_rs(1)))
+  coreMonitorBundle.rd0src := Mux(wb_ctrl.rxs1 || wb_ctrl.rfs1, wb_reg_inst(19,15), 0.U)
+  coreMonitorBundle.rd0val := Mux(wb_ctrl.rxs1 || wb_ctrl.rfs1, Reg(next=Reg(next=ex_rs(0))), 0.U)
+  coreMonitorBundle.rd1src := Mux(wb_ctrl.rxs2 || wb_ctrl.rfs2, wb_reg_inst(24,20), 0.U)
+  coreMonitorBundle.rd1val := Mux(wb_ctrl.rxs2 || wb_ctrl.rfs2, Reg(next=Reg(next=ex_rs(1))), 0.U)
```

Comment:
Same comment applies here.  Reading FP register is meaningful, so forcing values to 0 when there's no read gives a false sense that 0 means no read.